src/common/win32/errno.cc

```diff
@@ -46,6 +46,11 @@ __u32 ceph_to_hostos_errno_unsigned(__u32 r)
     case 9: return EBADF;
     case 10: return ECHILD;
     // same as EWOULDBLOCK
+    // need to be careful of treating EWOULDBLOCK same way as EAGAIN
+    // in errno.h (in Windows) the EWOULDBLOCK is set to 140, while EAGAIN is 11
```

Comment:
MDS doesn't work on windows IIRC but the clients can. Wondering how the client behaves which sends EWOULDBLOCK i.e. 140 while the MDS expects 11. In `include/fs_types.h` the `#define CEPHFS_EWOULDBLOCK     CEPHFS_EAGAIN` code works fine but not sure how this would go through. Maybe we can mimic this behaviour through a test case by injecting the errno? What do you think?

Reply:
Good point, will add test

Reply:
After deep diving I realize that this can never happen. There are 3 cases:

- New windows client <-> new MDS
will work because of the conversion while encodig/decoding

- Existing windows clients <-> new MDS version
existing Windows clients are using the `CEPHFS_*` codes, and will always send `11` because of `#define CEPHFS_EWOULDBLOCK CEPHFS_EAGAIN`. 

- New windows client <-> existing MDS version
new clients will use the new version of `MClientReply.h`, where the result value will be set to wire format on encoding (see `common/win32/errno.cc`):
```__u32 hostos_to_ceph_errno_unsigned(__u32 r) {
  // Windows errno -> Linux errno
  switch(r) {
   ...
    case EAGAIN: return 11;
    case EWOULDBLOCK: return 11;
  ...
}
```
it always converts `EWOULDBLOCK` to `EAGAIN`), meaning that on the wire there will be always 
11, which will be translated to `EAGAIN` in the MDS


Sorry for the long explanation, hope it helps!

Reply:
Awesome :) glad it isn't a blocker.